### ISSUE TITLE
Improve Graph pruning code

### ISSUE THREAD

**Issue by JavierCVilla** [vepadulano/PyRDF#15]
The current [implementation of the graph pruning](https://github.com/JavierCVilla/PyRDF/blob/master/PyRDF/Node.py#L165-L215) could be improved with a more robust logic. A simple refactoring of the code changes the result of some tests, pruning more than needed.

For example, a good refactoring to improve the readability of the code could be the following:

```python
    def _is_prunable():
        # Condition 1 - Not enough references
        # Explanation...
        if len(gc.get_referrers(self)) <= 3:
            return True

        # Condition 2 - Action value already computed
        # Explanation...
        if self.operation and self.operation.is_action() and self.value:
            return True

        return False

    def graph_prune(self):
        """
        ...docstrings...
        """
        children = []

        for n in self.children:
            # Select children based on pruning condition
            if not n.graph_prune():
                children.append(n)

        self.children = children

        if not self.children:
            return self._is_prunable()

        return False
```

However, this change increases the number of references to `self`, so the condition `len(gc.get_referrers(self)) <= 3` is not true any more.

**Comment by vepadulano** [comment]
The new implementation of the `Node` class brought by #28 follows these suggestions. This issue can be closed

**Comment by JavierCVilla** [comment]
Fixed in #28
